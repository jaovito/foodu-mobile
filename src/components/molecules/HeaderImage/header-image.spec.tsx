import {fireEvent, render, screen} from '@test-utils';
import {mock} from '@test/mocks';
import {HeaderImage} from '.';

describe('HeaderImage tests', () => {
  it('should be defined', () => {
    const headerImage = render(
      <HeaderImage address={mock.headerImage.address} />,
    );

    expect(headerImage).toBeDefined();
  });

  it('should be able to click into address button', async () => {
    render(<HeaderImage address={mock.headerImage.address} />);

    const button = await screen.findByText(mock.headerImage.address);
    fireEvent.press(button);

    expect(button).toBeTruthy();
  });

  it('should not be able to find the address', async () => {
    const {queryByText} = render(<HeaderImage />);

    const button = queryByText(mock.headerImage.address);

    expect(button).toBeFalsy();
  });

  it('should match with snapshot', async () => {
    const header = render(<HeaderImage />).toJSON();

    expect(header).toMatchSnapshot();
  });

  it('should match snapshot with an address', async () => {
    const header = render(
      <HeaderImage address={mock.headerImage.address} />,
    ).toJSON();

    expect(header).toMatchSnapshot();
  });
});
