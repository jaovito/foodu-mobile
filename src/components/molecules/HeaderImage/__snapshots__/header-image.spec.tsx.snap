// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderImage tests should match snapshot with an address 1`] = `
<View
  style={
    [
      {
        "height": 490,
      },
    ]
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      [
        {
          "alignItems": "center",
          "height": "100%",
          "justifyContent": "flex-end",
          "width": "100%",
        },
        {
          "alignItems": "flex-start",
          "justifyContent": "flex-start",
        },
      ]
    }
  >
    <Image
      resizeMode="cover"
      source={
        {
          "testUri": "../../../src/assets/header.png",
        }
      }
      style={
        [
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "height": "100%",
            "width": "100%",
          },
          undefined,
        ]
      }
    />
    <Text
      style={
        [
          {
            "color": "#0D0D0D",
            "fontFamily": "Poppins_400Regular",
            "fontSize": 31,
          },
          {
            "color": "#FFFFFF",
            "fontSize": 27,
            "marginLeft": 39,
            "marginTop": 147,
          },
        ]
      }
    >
      general.yourLocation
    </Text>
    <Text
      fontWeight="semiBold"
      numberOfLines={1}
      onPress={[Function]}
      style={
        [
          {
            "color": "#0D0D0D",
            "fontFamily": "Poppins_600SemiBold",
            "fontSize": 31,
          },
          {
            "color": "#FFFFFF",
            "fontSize": 29,
            "marginBottom": 31,
            "marginLeft": 39,
            "marginRight": 0,
            "marginTop": 16,
          },
        ]
      }
    >
      New York
    </Text>
    <Text
      fontWeight="semiBold"
      style={
        [
          {
            "color": "#0D0D0D",
            "fontFamily": "Poppins_600SemiBold",
            "fontSize": 31,
          },
          {
            "color": "#FFFFFF",
            "fontSize": 63,
            "marginLeft": 39,
          },
        ]
      }
    >
      general.orderFavoriteFood
    </Text>
  </View>
</View>
`;

exports[`HeaderImage tests should match with snapshot 1`] = `
<View
  style={
    [
      {
        "height": 490,
      },
    ]
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      [
        {
          "alignItems": "center",
          "height": "100%",
          "justifyContent": "flex-end",
          "width": "100%",
        },
        {
          "alignItems": "flex-start",
          "justifyContent": "flex-start",
        },
      ]
    }
  >
    <Image
      resizeMode="cover"
      source={
        {
          "testUri": "../../../src/assets/header.png",
        }
      }
      style={
        [
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "height": "100%",
            "width": "100%",
          },
          undefined,
        ]
      }
    />
    <Text
      style={
        [
          {
            "color": "#0D0D0D",
            "fontFamily": "Poppins_400Regular",
            "fontSize": 31,
          },
          {
            "color": "#FFFFFF",
            "fontSize": 27,
            "marginLeft": 39,
            "marginTop": 147,
          },
        ]
      }
    >
      general.yourLocation
    </Text>
    <Text
      fontWeight="semiBold"
      numberOfLines={1}
      onPress={[Function]}
      style={
        [
          {
            "color": "#0D0D0D",
            "fontFamily": "Poppins_600SemiBold",
            "fontSize": 31,
          },
          {
            "color": "#FFFFFF",
            "fontSize": 29,
            "marginBottom": 31,
            "marginLeft": 39,
            "marginRight": 0,
            "marginTop": 16,
          },
        ]
      }
    >
      general.city
    </Text>
    <Text
      fontWeight="semiBold"
      style={
        [
          {
            "color": "#0D0D0D",
            "fontFamily": "Poppins_600SemiBold",
            "fontSize": 31,
          },
          {
            "color": "#FFFFFF",
            "fontSize": 63,
            "marginLeft": 39,
          },
        ]
      }
    >
      general.orderFavoriteFood
    </Text>
  </View>
</View>
`;
